pkg/cmd/pulumi/deployment.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/pulumi/pulumi/pkg/v3/backend"
+	"github.com/pulumi/pulumi/pkg/v3/backend/display"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+	"github.com/spf13/cobra"
+)
+
+var stackDeploymentConfigFile string
+
+//nolint:unused // temporary ignore until we introduce the new commands
+func loadProjectStackDeployment(stack backend.Stack) (*workspace.ProjectStackDeployment, error) {
+	if stackDeploymentConfigFile == "" {
+		return workspace.DetectProjectStackDeployment(stack.Ref().Name().Q())
+	}
+	return workspace.LoadProjectStackDeployment(stackDeploymentConfigFile)
+}
+
+func saveProjectStackDeployment(psd *workspace.ProjectStackDeployment, stack backend.Stack) error {
+	if stackDeploymentConfigFile == "" {
+		return workspace.SaveProjectStackDeployment(stack.Ref().Name().Q(), psd)
+	}
+	return psd.Save(stackDeploymentConfigFile)
+}
+
+func newDeploymentCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		// This is temporarily hidden while we iterate over the new set of commands,
+		// we will remove before releasing these new set of features.
+		Hidden: true,
+		Use:    "deployment",
+		Short:  "Manage stack deployments on Pulumi Cloud",
+		Long: "Manage stack deployments on Pulumi Cloud.\n" +
+			"\n" +
+			"Use this command to trigger deployment jobs and manage deployment settings.",
+		Args: cmdutil.NoArgs,
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			return nil
+		}),
+	}
+
+	cmd.PersistentFlags().StringVar(
+		&stackDeploymentConfigFile, "config-file", "",
+		"Override the file name where the deployment settings are specified. Default is Pulumi.[stack].deploy.yaml")
+
+	cmd.AddCommand(newDeploymentSettingsCmd())
+
+	return cmd
+}
+
+func newDeploymentSettingsCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "settings",
+		Args:  cmdutil.ExactArgs(1),
+		Short: "Manages the stack's deployment settings",
```

Comment:
In general I'd try and keep these all active like the one you have above, so `Manage stack deployment settings` for instance.